### PR TITLE
Update python min version to 3.8

### PR DESCRIPTION
**Pull request by WilliamJamieson**
Addresses this [comment](https://github.com/spacetelescope/spherical_geometry/pull/218#issuecomment-1099616772) for @pllim.

### CODE REVIEW

**Comment by WilliamJamieson** [comment]
Note that the 3.7 required CI should be dropped in the repo settings.

File: setup.py [hunk 1]
```diff
@@ -136,6 +136,7 @@ def qd_config(arg):
         'astropy>=5.0.4',
         'numpy',
```
Comment (pllim) [inline setup.py:137]: To be consistent with doc.
```suggestion
        'numpy>=1.18',
```

File: CHANGES.rst [hunk 2]
```diff
@@ -9,6 +9,8 @@ Release Notes
 
 - Pin astropy min version to 5.0.4. [#218]
 
+- Set min python version to 3.8. [#219]
```
Comment (pllim) [inline CHANGES.rst:12]:
```suggestion
- Set Python min version to 3.8. [#219]
```

**Comment by pllim** [comment]
Also, I dropped 3.7 from branch protection rule. Thanks!

**Review by pllim — APPROVED** [review]
Thanks!

**Pull request merged**